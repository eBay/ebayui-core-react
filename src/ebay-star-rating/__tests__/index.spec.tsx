import React from 'react'
import { render } from '@testing-library/react';
import { EbayStarRating } from '../'

const htmlSnap =(ui: React.ReactElement): void => {
    const { asFragment } = render(ui);
    expect(asFragment()).toMatchSnapshot();
}
describe('star-rating-select', () => {
    it('renders defaults', async () => {
        await htmlSnap(<EbayStarRating value="3-5" />);
    });

    it('renders basic fieldset', async () => {
        await htmlSnap(<EbayStarRating />);
    });

    it('renders with 0 selected', async () => {
        await htmlSnap(<EbayStarRating value={0} a11yText="star rating"/> );
    });

    it('renders with 2 selected', async () => {
        await htmlSnap(<EbayStarRating value={2} a11yText="star rating"/>);
    });

    it('renders with 5 selected', async () => {
        await htmlSnap(<EbayStarRating value={5}/>);
    });
});
