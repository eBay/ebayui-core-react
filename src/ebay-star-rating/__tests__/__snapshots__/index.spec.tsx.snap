// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`star-rating-select renders basic fieldset 1`] = `
<DocumentFragment>
  <div
    class="star-rating"
    role="img"
  >
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`star-rating-select renders defaults 1`] = `
<DocumentFragment>
  <div
    class="star-rating"
    data-stars="3-5"
    role="img"
  >
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`star-rating-select renders with 0 selected 1`] = `
<DocumentFragment>
  <div
    aria-label="star rating"
    class="star-rating"
    data-stars="0"
    role="img"
  >
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`star-rating-select renders with 2 selected 1`] = `
<DocumentFragment>
  <div
    aria-label="star rating"
    class="star-rating"
    data-stars="2"
    role="img"
  >
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`star-rating-select renders with 5 selected 1`] = `
<DocumentFragment>
  <div
    class="star-rating"
    data-stars="5"
    role="img"
  >
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="star-rating__icon icon icon--star-dynamic"
      focusable="false"
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlink:href="#icon-star-dynamic"
      />
    </svg>
  </div>
</DocumentFragment>
`;
